p2p/__init__.py

```diff
@@ -602,44 +602,46 @@ def clone_content_item(self, slug, clone_slug, keep_embeds=False, keep_relateds=
                 }
                 payload['related_items'].append(related_item)
 
+        contributors = get_cloned_contributors(content_item)
+
+        if contributors:
+            del content_item['byline']
+            content_item['contributors'] = contributors
+
         # Clone the thing
         clone = self.create_content_item(payload)
         clone = clone.get('story', clone.get('html_story'))
 
         # if we have successfully cloned the content item, continue on
         if not clone.get('id'):
             raise P2PNotFound
-        else:
-            clone_contributors = []
-
-            # Split apart the byline string and iterate through it
-            if content_item.get('byline', None):
-                bylines = content_item.get('byline').split(',')
-                for byline in bylines:
-
-                    # Preemptively create a freeform contributor
-                    byline = byline.strip()
-                    byline_item = {"free_form_name": byline}
-
-                    # Search the contributors array for a matching adv byline
-                    for contributor in content_item.get('contributors'):
-                        # Wade through the nestedness
-                        contributor = contributor['contributor']
-                        if byline.lower() in contributor['title'].lower():
-                            # If a match was found, update the entry with the staff slug
-                            byline_item = {'slug': contributor['slug']}
-
-                    # Add the final result to the clone_contributors array
-                    clone_contributors.append(byline_item);
-
-            # If we have some contributors, let's add them
-            if len(clone_contributors) > 0:
-                try:
-                    self.append_contributors_to_content_item(clone['id'], clone_contributors)
-                except P2PException:
-                    raise
 
-            return clone['id']
+        return clone['id']
+
+    def get_cloned_contributors(self, content_item):
```

Comment:
Let's at least throw an underscore on the front of it and make it a private method. As someone who uses dir() to debug and figure out what the heck I can use in code, this kind of thing cornfuses me.

Mind throwing a docstring on it as well?

```
"""
Parses a content item payload's freeform bylines and advanced bylines
into a XXX
"""
```

^ Mind helping me write this?